package_bag/routines.py

```diff
@@ -21,29 +21,26 @@ class BagDiscoverer(object):
     """
 
     def run(self):
-        processed = []
-        unprocessed = self.discover_bags(settings.SRC_DIR)
-        for bag in unprocessed:
-            try:
-                bag_id = self.unpack_rename(bag, settings.TMP_DIR)
-                bag_path = join(settings.TMP_DIR, bag_id)
-                validate(bag_path)
-                bag_data = self.validate_metadata(bag_path)
-                new_bag = Bag.objects.create(
-                    original_bag_name=bag,
-                    bag_identifier=bag_id,
-                    bag_path=bag_path)
-                for key in ["Origin", "Rights-ID", "End-Date"]:
-                    setattr(new_bag, key.lower().replace("-", "_"), bag_data.get(key))
-                new_bag.save()
-                processed.append(bag_id)
-            except Exception as e:
-                remove_file_or_dir(bag_path)
-                print(e)
+        bag = self.discover_bags(settings.SRC_DIR)[0]
+        try:
+            bag_id = self.unpack_rename(bag, settings.TMP_DIR)
+            bag_path = join(settings.TMP_DIR, bag_id)
+            validate(bag_path)
+            bag_data = self.validate_metadata(bag_path)
+            new_bag = Bag.objects.create(
+                original_bag_name=bag,
+                bag_identifier=bag_id,
+                bag_path=bag_path)
+            for key in ["Origin", "Rights-ID", "End-Date"]:
+                setattr(new_bag, key.lower().replace("-", "_"), bag_data.get(key))
+            new_bag.save()
+        except Exception as e:
+            remove_file_or_dir(bag_path)
+            print(e)
         # what does this process bags function return? - you want to return something out of the view that indicates which objects were processed
         # e.g.: "{} bags discovered".format(len(processed)), processed
-        msg = "Bags discovered." if len(processed) else "No bags were found."
-        return msg, processed
+        msg = "Bags discovered." if bag else "No bags were found."
+        return msg, bag
```

Comment:
I think the issue here is that `bag` is an instance of the `Bag` class. Instead I think you want to return `bag_id`.